library/core/src/main/java/com/google/android/exoplayer2/trackselection/AdaptiveTrackSelection.java

```diff
@@ -540,7 +540,11 @@ public int evaluateQueueSize(long playbackPositionUs, List<? extends MediaChunk>
   @SuppressWarnings("unused")
   protected boolean canSelectFormat(
       Format format, int trackBitrate, float playbackSpeed, long effectiveBitrate) {
-    return Math.round(trackBitrate * playbackSpeed) <= effectiveBitrate;
+
+    boolean isNonIframeOnly = (format.roleFlags & C.ROLE_FLAG_TRICK_PLAY) == 0;
```

Comment:
Sure, that sounds fine.  Really this logic is just to prevent selecting these tracks ever, for now.  The right thing to do is add the logic to TrackSelection itself.  In our back-port to release we have subclassed `AdaptiveTrackSelection`.   The idea is when you move to high-speed mode (or scrub based trick-play) a TrackSelection occurs that results in a `TrackSelection.Definition` that includes only formats that are relevant (either all i-Frame or all non-iFrame tracks).   Oliver had other thoughts about adapting based on frame rate.   I think all of these are worth visiting, and the changes would be restricted to track selection so this code could be merged in first.

Reply:
Replaced with a positive logic variable, hopefully this is more clear.   The function should return fall if the track is an iFrame only track.

Logic to return true will depend on the trick-play implementation.